src/pytorch_lightning/accelerators/hpu.py

```diff
@@ -52,13 +55,27 @@ def get_parallel_devices(devices: int) -> List[torch.device]:
 
     @staticmethod
     def auto_device_count() -> int:
-        """Get the devices when set to auto."""
-        # TODO(@kaushikb11): Update this when api is exposed by the Habana team
-        return 8
+        """Returns the number of HPU devices when the devices is set to auto."""
+        try:
+            return torch_hpu.device_count()
+        except AttributeError:
+            return 0
```

Comment:
If not all versions have this and Habana is available, it should be 8, right?

Reply:
1 node contains 8 cards